src/IO/S3/copyS3File.h

```diff
@@ -19,9 +19,9 @@ using CreateReadBuffer = std::function<std::unique_ptr<SeekableReadBuffer>()>;
 
 /// Copies a file from S3 to S3.
 /// The same functionality can be done by using the function copyData() and the classes ReadBufferFromS3 and WriteBufferFromS3
-/// however copyS3File() is faster and spends less network traffic and memory.
+/// however copyS3FileNative() is faster and spends less network traffic and memory.
 /// The parameters `src_offset` and `src_size` specify a part in the source to copy.
-void copyS3File(
+void copyS3FileNative(
```

Comment:
We don't have to add this new interface function because `copyS3File` can check this new request setting and do the switch. I mean `CopyFileHelper::performCopy()` can call `copyDataToS3File()` like `CopyFileHelper::processCopyRequest()` does in case of some errors.

Reply:
There are two different functions for this because the signature is slightly different, pure-native does not need buffer creator, and someone may relay on this, though right now there is no such users in the code.

Reply:
[`5b34b16` (#51448)](https://github.com/ClickHouse/ClickHouse/pull/51448/commits/5b34b16dfb69f38cd76b27464ceec267537780f8)